ceremonies/src/lib.rs

```diff
@@ -1189,53 +1246,87 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
-	fn get_meetup_index(
+	fn get_participant_type(
 		community_ceremony: CommunityCeremony,
 		participant: &T::AccountId,
-	) -> Option<MeetupIndexType> {
-		let meetup_count = Self::meetup_count(community_ceremony);
-		let assignment_count = Self::assignment_counts(community_ceremony);
-
-		let assignment = Self::assignments(community_ceremony);
-
+	) -> Option<ParticipantType> {
 		if <BootstrapperIndex<T>>::contains_key(community_ceremony, &participant) {
-			let participant_index = Self::bootstrapper_index(community_ceremony, &participant) - 1;
-			if participant_index < assignment_count.bootstrappers {
-				return meetup_index(
-					participant_index,
-					assignment.bootstrappers_reputables,
-					meetup_count,
-				)
-			}
+			return Some(ParticipantType::Bootstrapper)
 		}
 		if <ReputableIndex<T>>::contains_key(community_ceremony, &participant) {
-			let participant_index = Self::reputable_index(community_ceremony, &participant) - 1;
-			if participant_index < assignment_count.reputables {
-				return meetup_index(
-					participant_index + assignment_count.bootstrappers,
-					assignment.bootstrappers_reputables,
-					meetup_count,
-				)
-			}
+			return Some(ParticipantType::Reputable)
 		}
-
 		if <EndorseeIndex<T>>::contains_key(community_ceremony, &participant) {
-			let participant_index = Self::endorsee_index(community_ceremony, &participant) - 1;
-			if participant_index < assignment_count.endorsees {
-				return meetup_index(participant_index, assignment.endorsees, meetup_count)
-			}
+			return Some(ParticipantType::Endorsee)
 		}
-
 		if <NewbieIndex<T>>::contains_key(community_ceremony, &participant) {
-			let participant_index = Self::newbie_index(community_ceremony, &participant) - 1;
-			if participant_index < assignment_count.newbies {
-				return meetup_index(participant_index, assignment.newbies, meetup_count)
-			}
+			return Some(ParticipantType::Newbie)
 		}
-
 		None
 	}
 
+	fn get_meetup_index(
+		community_ceremony: CommunityCeremony,
+		participant: &T::AccountId,
+	) -> Option<MeetupIndexType> {
+		let meetup_count = Self::meetup_count(community_ceremony);
+		let assignment_count = Self::assignment_counts(community_ceremony);
+
+		let assignment = Self::assignments(community_ceremony);
+
+		let participant_meetup_index: Option<MeetupIndexType>;
+		if let Some(participant_type) = Self::get_participant_type(community_ceremony, participant)
+		{
+			participant_meetup_index = match participant_type {
+				ParticipantType::Bootstrapper => {
+					let participant_index =
+						Self::bootstrapper_index(community_ceremony, &participant) - 1;
+					if participant_index < assignment_count.bootstrappers {
+						return meetup_index(
+							participant_index,
+							assignment.bootstrappers_reputables,
+							meetup_count,
+						)
+					}
+					None
+				},
```

Comment:
done. moving to a function would not make sense because you would also have to pass `assignment` and `assignment_count` to the function which would literally move all the functionality to the other function which seems pointless to me.